### PR TITLE
records: added record communities support.

### PR DESCRIPTION
**Pull request by alejandromumo**
closes https://github.com/zenodo/rdm-project/issues/648

### CODE REVIEW

File: inveniordm_py/records/resources.py [hunk 1]
```diff
@@ -297,3 +304,47 @@ def delete(self):
...
+
+        Usage:
+
+        .. code-block:: python
+
+            communities.add(["com1", "com2"])
+            communities.add("com1")
+            communities.add(RecordCommunityMetadata(communities=["com1", "com2"]))
+        """
+        _data = self._normalize_data(data)
```
Comment (alejandromumo) [inline inveniordm_py/records/resources.py:344]: This behavior is a bit different to other implementations.

E.g. for drafts the usage is `draft.update(data=DraftMetadata({...}))` (note that `data` is a `DraftMetadata` instance)

For communities, it felt better to allow `record.communities.add('blr')` (note that `blr` is just the slug of the community, however an instance of `RecordCommunityMetadata` can also be used)

File: inveniordm_py/records/metadata.py [hunk 2]
```diff
@@ -39,3 +40,37 @@ class RecordListMetadata(ListMetadata):
...
+
+    def to_request(self):
+        """Convert metadata to request body."""
+        return json.dumps(self._serialize_data())
+
+
+class RecordCommunitiesListMetadata(ListMetadata):
+    """Record communities list metadata class."""
+
+    item_class = RecordCommunityMetadata
```
Comment (jrcastro2) [inline inveniordm_py/records/metadata.py:76]: Comment: Having a `RecordCommunityMetadata` class that supports a list of communities looks a bit redundant with having `RecordCommunitiesListMetadata` wich supports a list of `RecordCommunityMetadata`. Could you explaing the use case of why we have both?

**Review by jrcastro2 — COMMENTED** [review]
Peer reviewed with @yashlamba 
Looks good, jsut left a comment on the `RecordCommunitiesListMetadata` which is not very clear to us.

File: inveniordm_py/records/metadata.py (hunk 2 above)
Comment (alejandromumo) [reply inveniordm_py/records/metadata.py:76]: In the client's design, `ListMetadata` objects support `search` methods (e.g. responses from the API that are generated from ES/OS). I.e., it implements `hits` and `aggregations` properties. Then, each object is a `RecordCommunityMetadata`. 

On the other hand, `Metadata` objects implement more generic representations of metadata. I.e. `RecordCommunityMetadata` refers to **one** record community.

**Pull request merged**